lib/logstash/inputs/s3.rb

```diff
@@ -286,7 +288,26 @@ def sincedb
 
   private
   def sincedb_file
-    File.join(ENV["HOME"], ".sincedb_" + Digest::MD5.hexdigest("#{@bucket}+#{@prefix}"))
+    digest = Digest::MD5.hexdigest("#{@bucket}+#{@prefix}")
+    dir = File.join(LogStash::SETTINGS.get_value("path.data"), "plugins", "inputs", "s3")
+    FileUtils::mkdir_p(dir)
+    path = File.join(dir, "sincedb_#{digest}")
+
+    # Migrate old default sincedb path to new one.
+    if ENV["HOME"]
+      old = File.join(ENV["HOME"], ".sincedb_" + Digest::MD5.hexdigest("#{@bucket}+#{@prefix}"))
```

Comment:
@jordansissel we can just reuse `digest` here can't we?

Reply:
I kind of left this one in case we change the new (in path.data) digest method in the future; it would avoid accidental bugs by keeping this "old" one not reused.

Reply:
@jordansissel fair point :)

Reply:
I added a comment above this line now to try to explain the reasoning for our future selves.